### PR TITLE
ValidationPage: SMAPI integration

### PR DESCRIPTION
**Pull request by chris-ramon**
#### Overview
- ValidationPage:
  - [x] Adds & wires `Vendor ID` ui.
  - [x] Finds and sends `smAPIAccessToken` when validating scripts.
  - [x] Saves `vendorID` to firebase.
  - [x] Loads and sets existing `vendorID`.
- Closes https://github.com/bespoken/dashboard/issues/464 & https://github.com/bespoken/dashboard/issues/465

output

#### Test plan
- [x] Manual test
  - [x] Vendor ID is save and set on validation page load: <details><summary>GIF</summary></details>
  - [x] Vendor ID if invalid for SMAPI, shows proper error message:<details><summary>GIF</summary></details>
  - [x] Form, for unknown errors, shows proper message error:<details><summary>GIF</summary></details>

#### Action required
- Merge along `source-api` PR: https://github.com/bespoken/source-api/pull/33

### CODE REVIEW

File: src/models/user.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 export class UserDetails {
     readonly silentEchoToken: string;
-    constructor(silentEchoToken: string) {
+    readonly smAPIAccessToken: string;
+    constructor(silentEchoToken: string, smAPIAccessToken: string) {
         this.silentEchoToken = silentEchoToken;
+        this.smAPIAccessToken = smAPIAccessToken;
```
Comment (jkelvie) [inline src/models/user.ts]: We are essentially going to deprecate the silent echo token, correct? New users will not need both.

Comment (jkelvie) [reply src/models/user.ts]: I would also call it SMAPIAccessToken - consistent with how Amazon does it.

Comment (chris-ramon) [reply src/models/user.ts]: > We are essentially going to deprecate the silent echo token, correct? New users will not need both.

Well, we can preserve both, they clearly different tokens, one for virtual device, one for SMAPI.

Also preserving both will allow us to do no more changes on silent-echo side for SMAPI integration, more than the `/refresh_token` endpoint which is working great.

> I would also call it SMAPIAccessToken - consistent with how Amazon does it.

Good call!, I'll rename it.

Comment (jkelvie) [reply src/models/user.ts]: They are actually the same token - the one token covers both sets of privileges. That's why I was saying we don't need both.

**Comment by chris-ramon** [comment]
> Can we put the vendor ID and token fields on the same line next to each other? They are taking up too much space now

Good call!, addressed on: https://github.com/bespoken/dashboard/pull/466/commits/0bca9da5ee59c66937f68f0a251b25ce9b8def68


image

**Pull request merged**